engine/gamesys/src/gamesys/scripts/script_window.cpp

```diff
@@ -180,6 +180,13 @@ static void RunCallback(CallbackInfo* cbinfo)
 static int SetListener(lua_State* L)
 {
     WindowInfo* window_info = &g_Window;
+
+    luaL_checkany(L, 1);
```

Comment:
Do we need to check both luaL_checkany() and lua_isnoneornil()? Is it not sufficient to use lua_isnoneornil()?

Reply:
Or I guess it depends on what is acceptable?

```
-- should both of these clear the listener?
window.set_listener()
window.set_listener(nil)
```

Reply:
>should both of these clear the listener?

Feels like the answer should be yes. What else would the argument less function call do?

Reply:
As a user I would probably want an error if I forget to pass a listener to `set_listener`. But I think it is up to you to decide :)

Reply:
So I guess I should either:
- do `luaL_checkany()` and change `lua_isnoneornil()` to `lua_isnil()`
- or remove `luaL_checkany()`

so do we want `window.set_listener()` to unsubscribe or to throw an error?

Reply:
It feels just as fair to me personally to only warn the user that not giving an argument does nothing.

Reply:
Let's go with:

window.set_listener() -- error!
window.set_listener(null) -- clear listener

Reply:
Thanks, I will close PR, change `lua_isnoneornil()` to `lua_isnil()`, rename branch and reopen pull request.

Reply:
No need to close the PR!

Reply:
Make the changes here instead. That way we keep the history and our discussions intact.

Reply:
ok :) there is just no way to rename a branch without closing it

Reply:
@britzl done :)